File: python-app/media_store.py
import uuid

import config

config.MEDIA_DIR.mkdir(parents=True, exist_ok=True)


def _type_for(message):
    if message.photo:
        return 'photo'
    if message.video:
        return 'video'
    doc = message.document
    if doc and doc.mime_type:
        if doc.mime_type.startswith('video/'):
            return 'video'
        if doc.mime_type.startswith('image/'):
            return 'photo'
    return None  # skip generic documents/audio - keep map/feed focused on visual media


def _ext_for(message, media_type):
    doc = message.document
    if doc:
        for attr in doc.attributes:
            file_name = getattr(attr, 'file_name', None)
            if file_name and '.' in file_name:
                return file_name.rsplit('.', 1)[-1].lower()
    return 'mp4' if media_type == 'video' else 'jpg'


async def save_media(client, message):
    media_type = _type_for(message)
    if not media_type:
        return None

    ext = _ext_for(message, media_type)
    filename = f'{uuid.uuid4()}.{ext}'
    path = config.MEDIA_DIR / filename

    await client.download_media(message, file=str(path))

    return {'type': media_type, 'url': f'/media/{filename}'}

File: server/src/mediaStore.js
const fs = require('fs');
const path = require('path');
const crypto = require('crypto');
const config = require('./config');

fs.mkdirSync(config.mediaDir, { recursive: true });

function typeFor(message) {
  if (message.photo) return 'photo';
  if (message.video) return 'video';
  const mime = message.document && message.document.mimeType;
  if (mime && mime.startsWith('video/')) return 'video';
  if (mime && mime.startsWith('image/')) return 'photo';
  return null; // skip generic documents/audio - keep map/feed focused on visual media
}

function extFor(message, type) {
  const attrs = (message.document && message.document.attributes) || [];
  const fileNameAttr = attrs.find((a) => a.fileName);
  if (fileNameAttr && fileNameAttr.fileName.includes('.')) {
    return fileNameAttr.fileName.split('.').pop().toLowerCase();
  }
  return type === 'video' ? 'mp4' : 'jpg';
}

async function saveMedia(client, message) {
  const type = typeFor(message);
  if (!type) return null;

  const ext = extFor(message, type);
  const filename = `${crypto.randomUUID()}.${ext}`;
  const outputFile = path.join(config.mediaDir, filename);

  await client.downloadMedia(message, { outputFile });

  return { type, url: `/media/${filename}` };
}

module.exports = { saveMedia };
